stacker/src/expull.rs

```diff
@@ -132,9 +142,28 @@ impl ExpUnrolledLinkedList {
         if addr.is_null() {
             return;
         }
-        let last_block_len = get_block_size(self.block_num) as usize - self.remaining_cap as usize;
 
-        // Full Blocks
+        // Calculate last block length with bounds checking to prevent underflow
+        let block_size = get_block_size(self.block_num) as usize;
+        let last_block_len = block_size.saturating_sub(self.remaining_cap as usize);
+
+        // Safety check: if remaining_cap > block_size, the metadata is corrupted
+        // This should never happen with correct usage, so panic to catch bugs early
+        assert!(
+            self.remaining_cap as usize <= block_size,
+            "ExpUnrolledLinkedList metadata corruption detected: remaining_cap ({}) > block_size \
+             ({}). This indicates a serious bug, possibly from: 1. Copying ExpUnrolledLinkedList \
+             and using stale metadata 2. Race condition in parallel processing 3. Memory \
+             corruption. block_num={}, head={:?}, tail={:?}",
```

Comment:
```suggestion
             ({}). This indicates a serious bug, please report! (block_num={}, head={:?}, tail={:?})",
```

What parallel processing?

Reply:
> What parallel processing?
multi-threaded execution.

Applied your suggestion.